library/aws/policies_v2.ts

```diff
@@ -0,0 +1,326 @@
+import * as AR from "../../providers/aws/resources.ts";
+import { ArnSecret } from "./secrets.ts";
+
+export interface NamedPolicy {
+  name: string;
+  policy: {
+    Statement: Statement[],
+    [key: string]: unknown;
+  };
+}
+
+export interface Statement {
+  Action: unknown;
+  Effect: unknown;
+  [key: string]: unknown;
+}
+
+export function newNamedPolicy(
+  name: string,
+  statements: Statement[],
+): NamedPolicy {
+  return {
+    name,
+    policy: {
+      Version: "2012-10-17",
+      Statement: statements,
+    },
+  };
+}
+
+export function combineNamedPolicies(
+  name: string,
+  namedPolicies: NamedPolicy[],
+): NamedPolicy {
+  const Statement: Statement[] = [];
+  for (const np of namedPolicies) {
+    Statement.push(...np.policy.Statement);
+  }
+
+  return {
+    name,
+    policy: {
+      Version: "2012-10-17",
+      Statement,
+    },
+  };
+}
+
+
+export function assumeServiceRole(Service: string): Statement[] {
+  return [
+    {
+      Action: "sts:AssumeRole",
+      Principal: { Service },
+      Effect: "Allow",
+      Sid: "",
+    },
+  ]
+}
+
+export function publishMetrics(): Statement[] {
+  return [
+    {
+      Action: [
+        "cloudwatch:GetMetricStatistics",
+        "cloudwatch:ListMetrics",
+        "cloudwatch:PutMetricData",
+        "ec2:DescribeTags",
+      ],
+      Effect: "Allow",
+      Resource: "*",
+    },
+  ];
+}
+
+export function s3ReadonlyBuckets(
+  buckets: string[],
+  key_prefix: string = "*",
+): Statement[] {
+  return [
+    {
+      Effect: "Allow",
+      Action: ["s3:ListBucket"],
+      Resource: buckets.map((bucket) => `arn:aws:s3:::${bucket}`),
+    },
+    {
+      Action: ["s3:GetObject"],
+      Effect: "Allow",
+      Resource: buckets.map((bucket) => `arn:aws:s3:::${bucket}/${key_prefix}`),
+    },
+  ];
+}
+
+export function s3PublicReadonly(
+  bucket: string,
+  key_prefix: string = "*",
+): Statement[] {
+  return [
+    {
+      Action: ["s3:GetObject"],
+      Effect: "Allow",
+      Principal: "*",
```

Comment:
The principals are only for trust-policies or resource-policies, it could lead to an error of trying to use this generated read only as a statement for a identity policy

Reply:
This is the existing policy, moved out of `policies.ts`. Are you saying we should better comment this to reflect it's purpose? Or actually change it and affect live policies on AWS?